packages/dids/src/types/did-core.ts

```diff
@@ -339,6 +339,11 @@ export type DidResolutionMetadata = {
    */
   error?: string;
 
+  /**
+   * Signals whether or not the resolved DID document is stale in the cache.
+   */
+  isStaleInCache?: boolean;
```

Comment:
Could we come up with a way of implementing this state which doesn't involve adding a property to `DidResolutionMetadata` which isn't in the specification data model https://www.w3.org/TR/did-core/#did-resolution-metadata?

Reply:
Thanks! Great call out!

I extended the `DidResolutionResult` as `CachedDidResolutionResult` and put that property within there.

Let me know what you think about this approach.

Reply:
yeah this is great 👍